hazelcast-jet-core/src/main/java/com/hazelcast/jet/stream/impl/reducers/GroupingSinkReducer.java

```diff
@@ -16,52 +16,53 @@
 
 package com.hazelcast.jet.stream.impl.reducers;
 
-import com.hazelcast.core.IMap;
 import com.hazelcast.jet.DAG;
+import com.hazelcast.jet.Distributed.Function;
+import com.hazelcast.jet.ProcessorSupplier;
 import com.hazelcast.jet.Vertex;
-import com.hazelcast.jet.stream.DistributedCollector.Reducer;
+import com.hazelcast.jet.stream.DistributedCollector;
 import com.hazelcast.jet.stream.impl.pipeline.Pipeline;
 import com.hazelcast.jet.stream.impl.pipeline.StreamContext;
 import com.hazelcast.jet.stream.impl.processor.CombineGroupsP;
 import com.hazelcast.jet.stream.impl.processor.GroupAndAccumulateP;
 
-import java.util.function.Function;
 import java.util.stream.Collector;
 
 import static com.hazelcast.jet.Edge.between;
 import static com.hazelcast.jet.KeyExtractors.entryKey;
 import static com.hazelcast.jet.Partitioner.HASH_CODE;
-import static com.hazelcast.jet.Processors.writeMap;
 import static com.hazelcast.jet.stream.impl.StreamUtil.executeJob;
 
-public class GroupingIMapReducer<T, A, K, D> implements Reducer<T, IMap<K, D>> {
+public class GroupingSinkReducer<T, A, K, D, R> implements DistributedCollector.Reducer<T, R> {
 
-    private final String mapName;
+    private final Function<StreamContext, ? extends R> targetFunction;
     private final Function<? super T, ? extends K> classifier;
     private final Collector<? super T, A, D> collector;
+    private final ProcessorSupplier processorSupplier;
 
-    public GroupingIMapReducer(String mapName, Function<? super T, ? extends K> classifier,
-                                Collector<? super T, A, D> collector) {
-        this.mapName = mapName;
+    public GroupingSinkReducer(Function<StreamContext, ? extends R> targetFunction,
+                               Function<? super T, ? extends K> classifier,
+                               Collector<? super T, A, D> collector,
+                               ProcessorSupplier processorSupplier) {
+        this.targetFunction = targetFunction;
         this.classifier = classifier;
         this.collector = collector;
+        this.processorSupplier = processorSupplier;
     }
 
     @Override
-    public IMap<K, D> reduce(StreamContext context, Pipeline<? extends T> upstream) {
-        IMap<K, D> target = context.getJetInstance().getMap(mapName);
-
+    public R reduce(StreamContext context, Pipeline<? extends T> upstream) {
         DAG dag = new DAG();
         Vertex previous = upstream.buildDAG(dag);
         Vertex merger = dag.newVertex("group-and-accumulate",
                 () -> new GroupAndAccumulateP<>(classifier, collector));
         Vertex combiner = dag.newVertex("combine-groups", () -> new CombineGroupsP<>(collector));
-        Vertex writer = dag.newVertex("write-map-" + mapName, writeMap(mapName));
+        Vertex writer = dag.newVertex("write", processorSupplier);
```

Comment:
map name should be in the Vertex, so the vertex name should be passed in as well.